Assets/Scripts/DayNightCycle/DayAndNightControl.cs

```diff
@@ -0,0 +1,246 @@
+            currentTime = (float) GameSettings.Time.currentTime; 
+            //additional check to ensure MC cant set non-MC daylengths to less than 60
+            if ((float)GameSettings.Time.dayLength < 60)
+            {
+                DayLength = (float)GameSettings.Time.dayLength;
+            }
+            else { DayLength = 60; }
+            pause = (bool) GameSettings.Time.pause;
+
+            if (!pause)
+            {
+                // 300s since MC updated this
+                var diff = (float) (DateTime.UtcNow - settings.LastModified).TotalSeconds;
+                // 5 += ( 300 / 300) * 24 => 29 % 24 => 5
+                currentTime += (diff / DayLength) * 24;
+                //If time passed will put the currentTime over 24, do maths to correct for this and give an accurate
+                //time according to a 24h time range
+                if (currentTime > 24)
+                {
+                    currentTime = currentTime - (24 * (int)Math.Floor(currentTime/24));
+                }
+            }
+        }
+
+        private void OnDestroy()
+        {
+            Service.Settings.OnTimeSettingsChanged -= Settings_OnTimeSettingsChanged;
+        }
+
+        // Update is called once per frame
+        void Update()
+        {
+            
+            // This is to prevent null reference errors because non-MCs needs a few frames before the camera will be available.
+            if (MainCamera == null)
+            {
+                MainCamera = Camera.main;
+            }
+            else
+            {
+                MoonCamera.fieldOfView = MainCamera.fieldOfView;
+            }
+
+            if (!pause)
+            {
+                currentTime += (Time.deltaTime / DayLength) * 24;
+                if (CurrentTime01 >= 1) // If CurrentTime >= 24 hours
+                {
+                    currentTime = 0; // Reset time back to midnight
+                    currentDay++; // Increment the day counter
+                }
+            }
+
+            if (frames == lightingUpdateInterval)
+            {
+                frames = 0;
+                UpdateLight();
+                UpdateMaterial();
+            }
+            frames++;
+            //MC loads settings
+            if (PhotonNetwork.isMasterClient)
+            {
+                GameSettings.Time.currentTime = currentTime;
+                GameSettings.Time.dayLength = DayLength;
+                GameSettings.Time.pause = pause;
+            }
+        }
+
+        void UpdateMaterial()
+        {
+            skyBoxPROCEDURAL.SetVector("_Axis", directionalLight.transform.right);
+            skyBoxPROCEDURAL.SetFloat("_Angle", -CurrentTime01 * 360f);
+        }
+
+        void UpdateLightingSettings()
+        {
+            RenderSettings.skybox = skyBoxPROCEDURAL;
+            RenderSettings.sun = directionalLight; // Procedural skybox needs this to work
+            RenderSettings.fog = true;
+
+            if (timecycle)
```

Comment:
if statement can be inverted to make the code more readable => `if (!timeCycle) return;`